src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/transfercase/BulkPrintTransferService.java

```diff
@@ -0,0 +1,115 @@
+        this.caseSearchApi = caseSearchApi;
+        this.userService = userService;
+        this.caseMapper = caseMapper;
+        this.transferCaseDocumentPublishService = transferCaseDocumentPublishService;
+        this.transferCaseNotificationsService = transferCaseNotificationsService;
+        this.coreCaseDataService = coreCaseDataService;
+    }
+
+    public void findCasesAndTransfer() {
+        User user = userService.authenticateAnonymousCaseWorker();
+        String authorisation = user.getAuthorisation();
+        List<Claim> claimsReadyForTransfer = caseSearchApi.getClaimsReadyForTransfer(user);
+        Map<Claim, CCDCase> ccdCaseMap = new HashMap<>();
+        claimsReadyForTransfer.forEach(claim -> ccdCaseMap.put(claim, caseMapper.to(claim)));
+        ccdCaseMap.forEach((claim, ccdCase) -> {
+            CCDCase ccdCaseTransferred = transferCase(ccdCase, claim, authorisation);
+            CCDCase ccdCaseUpdated =  updateTransferContent(ccdCaseTransferred);
+            updateCaseInCCD(ccdCaseUpdated, authorisation);
+        });
+    }
+
+    public CCDCase transferCase(CCDCase ccdCase, Claim claim, String authorisation) {
+        CCDCase updated = transferCaseDocumentPublishService.publishCaseDocuments(ccdCase, authorisation, claim);
+        sendEmailNotifications(updated, claim);
+        return updateCaseData(updated);
+    }
+
+    private CCDCase updateTransferContent(CCDCase ccdCase) {
+        return ccdCase.toBuilder()
+            .transferContent(ccdCase.getTransferContent().toBuilder()
+                .transferCourtName(ccdCase.getHearingCourtName())
+                .transferCourtAddress(ccdCase.getHearingCourtAddress())
+                .transferReason(CCDTransferReason.OTHER)
+                .transferReasonOther(REASON).build())
+            .build();
+    }
+
+    private CCDCase updateCaseData(CCDCase ccdCase) {
+
+        CCDTransferContent transferContent;
+        if (ccdCase.getTransferContent() != null) {
+            transferContent = ccdCase.getTransferContent().toBuilder()
+                .dateOfTransfer(LocalDate.now()).build();
+        } else {
+            transferContent = CCDTransferContent.builder()
+                .dateOfTransfer(LocalDate.now()).build();
+        }
+        return ccdCase.toBuilder()
+            .transferContent(transferContent)
+            .build();
+
+    }
+
+    private void sendEmailNotifications(CCDCase ccdCase, Claim claim) {
+
+        transferCaseNotificationsService.sendClaimUpdatedEmailToClaimant(claim);
+
+        if (isDefendantLinked(ccdCase)) {
+            transferCaseNotificationsService.sendClaimUpdatedEmailToDefendant(claim);
+        }
+    }
+
+    private boolean isDefendantLinked(CCDCase ccdCase) {
+        return !StringUtils.isBlank(ccdCase.getRespondents().get(0).getValue().getDefendantId());
+    }
+
+    private void updateCaseInCCD(CCDCase ccdCase, String authorisation) {
+        ccdCase.setState(ClaimState.TRANSFERRED.getValue());
```

Comment:
State is not serialised when converting a CcdCase to json so this line is effectively useless. The state will be set by your event.